cmd/nerdctl/run_linux_test.go

```diff
@@ -247,7 +247,7 @@ func TestRunWithFluentdLogDriverWithLogOpt(t *testing.T) {
 	time.Sleep(3 * time.Second)
 
 	testContainerName := containerName + "test"
-	base.Cmd("run", "-d", "--log-driver", "fluentd", "--log-opt", "fluentd-address=127.0.0.1:24225",
+	base.Cmd("run", "-d", "--log-driver", "fluentd", "--log-opt", "fluentd-address=127.0.0.1:24225", "--log-opt", "fluentd-log-to-json=true",
```

Comment:
Thanks for the change. I am not sure why this feature is required:
1. It is already supported by fluentd as form of multiple outputs (https://docs.fluentd.org/output/secondary_file, https://docs.fluentd.org/output/file). Is there a use-case?
2. It is not supported in docker.
3. Even stdout is supported by fluentd (https://docs.fluentd.org/output/stdout) whichis what this test is doing.

if you want to still merge the change and maintainers are fine, can you update the readme.

Reply:
Sorry for replying late

1. Help people to use `nerdctl logs`  to get the container log(quickly debug, not need to go to the fluent backend server to get the log) 
2. Yep, this is a nerdctl-only feature
3. Yep, but in common use case, the people cannot decide whether the fluentd use stdout or not

Reply:
Could this PR be more generic then (i.e for all loggers except default and json)? There would be other remote log forwarders that can make use of this functionality.

Reply:
emmmm, I want to split it into different PR,https://github.com/containerd/nerdctl/issues/1374

Reply:
Hmm! Would you copy the same code in every logger then? Is it possible to make the code generic for any logger and have it in a single place so that when new loggers are added; they automatically inherit this logic rather than have more changes in the code base.